src/main/scala/io/shiftleft/fuzzyc2cpg/Fuzzyc2Cpg.scala

```diff
@@ -0,0 +1,139 @@
+package io.shiftleft.fuzzyc2cpg
+
+import java.nio.file.Path
+
+import io.shiftleft.codepropertygraph.generated.Languages
+import io.shiftleft.fuzzyc2cpg.Utils.{getGlobalNamespaceBlockFullName, newEdge, newNode}
+import io.shiftleft.fuzzyc2cpg.output.protobuf.OutputModuleFactory
+import io.shiftleft.proto.cpg.Cpg.{CpgStruct, NodePropertyName}
+import io.shiftleft.proto.cpg.Cpg.CpgStruct.Edge.EdgeType
+import io.shiftleft.proto.cpg.Cpg.CpgStruct.Node
+import io.shiftleft.proto.cpg.Cpg.CpgStruct.Node.NodeType
+import org.slf4j.LoggerFactory
+import io.shiftleft.fuzzyc2cpg.Utils._
+import io.shiftleft.fuzzyc2cpg.output.CpgOutputModuleFactory
+import io.shiftleft.fuzzyc2cpg.parser.modules.AntlrCModuleParserDriver
+
+class Fuzzyc2Cpg[T](outputModuleFactory : CpgOutputModuleFactory[T]) {
+
+  def this(outputPath : String) = {
+    this(new OutputModuleFactory(outputPath, true, false)
+      .asInstanceOf[CpgOutputModuleFactory[T]])
+  }
+
+  def runAndOutput(fileAndDirNames: Array[String]) = {
+    val inputPaths = fileAndDirNames
+    val sourceFileNames = SourceFiles.determine(inputPaths.toList).sorted
+
+    val filenameToNamespaceBlock = createStructuralCpg(sourceFileNames, outputModuleFactory)
+
+    // TODO improve fuzzyc2cpg namespace support. Currently, everything
+    // is in the same global namespace so the code below is correctly.
+    filenameToNamespaceBlock.par.foreach(createCpgForCompilationUnit)
+    outputModuleFactory.persist()
+  }
+
+  private def createStructuralCpg(filenames: List[String], cpgOutputModuleFactory: CpgOutputModuleFactory[T]):
+    List[(String, CpgStruct.Node)] = {
+
+    def addMetaDataNode(cpg : CpgStruct.Builder): Unit = {
+      val metaNode = newNode(NodeType.META_DATA)
+        .addStringProperty(NodePropertyName.LANGUAGE, Languages.C)
+        .build
+      cpg.addNode(metaNode)
+    }
+
+    def addAnyTypeAndNamespaceBlock(cpg : CpgStruct.Builder): Unit = {
+      val globalNamespaceBlockNotInFileNode = createNamespaceBlockNode(None)
+      cpg.addNode(globalNamespaceBlockNotInFileNode)
+    }
+
+    def createFileNode(pathToFile: Path): Node = {
+      newNode(NodeType.FILE)
+        .addStringProperty(NodePropertyName.NAME, pathToFile.toString)
+        .build()
+    }
+
+    def createFilesAndNamespaceBlocks(cpg: CpgStruct.Builder) : List[(String, CpgStruct.Node)] =
+      filenames.map{filename =>
+        val pathToFile = new java.io.File(filename).toPath
+        val fileNode = createFileNode(pathToFile)
+        val namespaceBlock = createNamespaceBlockNode(Some(pathToFile))
```

Comment:
So there is one node per file it seems 🤔Am I correct?

Reply:
yes. The namespace handling is kind of fake in fuzzyc2cpg at the moment. They're all just in the global namespace.